app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationViewActivity.java

```diff
@@ -310,27 +308,26 @@ private boolean getShouldSimulateRoute() {
     return sharedPreferences.getBoolean(getString(R.string.simulate_route_key), false);
   }
 
+  private String getRouteProfile() {
+    SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+    return sharedPreferences.getString("route_profile", "");
+  }
+
   private void launchNavigationWithRoute() {
-    Locale locale = getLocale();
-    MapboxNavigationOptions navigationOptions =
-      MapboxNavigationOptions.builder()
-        .locale(locale)
-        .unitType(getUnitType())
-        .build();
+    if (route == null) {
+      Snackbar.make(mapView, "Current route is not available", Snackbar.LENGTH_SHORT).show();
+      return;
+    }
 
-    NavigationViewOptions.Builder optionsBuilder = NavigationViewOptions.builder()
+    Locale locale = getLocale();
+    NavigationLauncherOptions.Builder optionsBuilder = NavigationLauncherOptions.builder()
       .shouldSimulateRoute(getShouldSimulateRoute())
-      .navigationOptions(navigationOptions);
```

Comment:
Are you sure it's safe to remove this logic? This was for when the user changes language, so you fetch a new route.